.github/workflows/release.yml

```diff
@@ -61,3 +61,22 @@ jobs:
         if: steps.changesets.outputs.published == 'true'
         # Do something more interesting when a publish happens.
         run: echo ${{ steps.changesets.outputs.publishedPackages }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ vars.GHCR_USER }}
+          password: ${{ secrets.GHCR_TOKEN }}
+
+      - name: Build Docker Image
+        run: |
+          docker build -t ghcr.io/kadena.js/graph:latest .
+          cd packages/apps/graph && docker compose build graphql-source -t ghcr.io/kadena/graph:latest -t ghcr.io/kadena/graph:preview
+
+      - name: Push Docker Image to GHCR
+        run: |
+          docker push ghcr.io/kadena.js/graph:latest && docker push ghcr.io/kadena/graph:preview
```

Comment:
We want to publish `:latest` when `steps.changesets.outputs.publishedPackages` contains `@kadena/graph`.

We want to publish `:preview` when there are changes to `packages/apps/graph/**`

```suggestion
      - name: Build Docker Image for `:latest`
        id: docker-build-graph-latest
        # only run when `changesets` has published for `@kadena/graph`
        if: steps.changesets.outputs.published == 'true' && contains(steps.changesets.outputs.publishedPackages, '@kadena/graph')
        run: |
          cd packages/apps/graph && docker compose build graphql -t ghcr.io/kadena/graph:latest -t ghcr.io/kadena/graph:$(npm view @kadena/graph --json |jq '."dist-tags".latest')

      - name: Push Docker Image to GHCR for `:latest`
        # only publish when above is success
        if: steps.docker-build-graph-latest.conclusion == 'success'
        run: |
          docker push ghcr.io/kadena.js/graph:latest && docker push ghcr.io/kadena/graph:latest && docker push ghcr.io/kadena/graph:$(npm view @kadena/graph --json |jq '."dist-tags".latest')
```


And we have to add for `:preview`:

```
      - name: Build Docker Image `:preview`
        id: docker-build-graph-preview
        # no need to publish when `:latest` is publishing
        if: steps.docker-build-graph-latest.conclusion != 'skipped'
        # only run when `graph` has changes
        on:
		  push:
		    paths:
		      - 'packages/apps/graph/**'
        run: |
          cd packages/apps/graph && docker compose build graphql-source -t ghcr.io/kadena/graph:preview -t ghcr.io/kadena/graph:${GITHUB_SHA::7}
      - name: Push Docker Image to GHCR for `:preview`
        if: steps.docker-build-graph-preview.conclusion == 'success'
        run: |
          docker push ghcr.io/kadena/graph:preview && docker push ghcr.io/kadena/graph:${GITHUB_SHA::7}
```